### PR TITLE
feat: steps progress

### PR DESCRIPTION
**Pull request by 07akioni**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution
image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |       Steps support `percent` prop to show step icon progress.    |
| 🇨🇳 Chinese |        Steps 新增 `percent` 属性来展示步骤图标的进度。   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/steps/demo/progress-debug.md](https://github.com/ant-design/ant-design/blob/feat-steps-progress/components/steps/demo/progress-debug.md)
[View rendered components/steps/demo/progress.md](https://github.com/ant-design/ant-design/blob/feat-steps-progress/components/steps/demo/progress.md)
[View rendered components/steps/index.en-US.md](https://github.com/ant-design/ant-design/blob/feat-steps-progress/components/steps/index.en-US.md)
[View rendered components/steps/index.zh-CN.md](https://github.com/ant-design/ant-design/blob/feat-steps-progress/components/steps/index.zh-CN.md)

### CODE REVIEW

File: components/steps/index.en-US.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ The whole of the step bar.
 | status | To specify the status of current step, can be set to one of the following values: `wait` `process` `finish` `error` | string | `process` |  |
 | initial | Set the initial step, counting from 0 | number | 0 |  |
 | onChange | Trigger when Step is changed | (current) => void | - |  |
+| percentage | Progress circle percentage of current step in `process` status | number | - | 4.5.0 |
+| progressIcon | Render step icon with percentage | ({node, index, status, title, description, percentage}) => ReactNode | - | 4.5.0 |
```
Comment (zombieJ) [inline components/steps/index.en-US.md:42]: `progressIcon` 暂时先不开放给用户，有 `percentage` 就行了。

Comment (zombieJ) [reply components/steps/index.en-US.md:42]: 参考 Progress 组件，用 `percent`

File: package.json [hunk 2]
```diff
@@ -134,7 +134,7 @@
     "rc-resize-observer": "^0.2.3",
     "rc-select": "^11.0.10",
     "rc-slider": "~9.3.0",
-    "rc-steps": "~4.0.1",
+    "rc-steps": "^4.1.0",
```
Comment (afc163) [inline package.json:137]:
```suggestion
    "rc-steps": "~4.1.0",
```

File: components/steps/demo/progress.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import { Steps, Popover } from 'antd';
+
+const { Step } = Steps;
+
+const progressIconRender = ({ node, percentage }) => {
+  return <Popover content={`${percentage}%`}>{node}</Popover>;
+};
+
+ReactDOM.render(
+  <Steps current={1} percentage={60} progressIcon={progressIconRender}>
```
Comment (afc163) [inline components/steps/demo/progress.md:26]: 用一个属性来收 `progress={{ ... }}`。

File: components/steps/index.en-US.md (hunk 1 above)
Comment (07akioni) [reply components/steps/index.en-US.md:42]: 第二个他们文档里面需要，就先不给？

Comment (07akioni) [reply components/steps/index.en-US.md:42]: 讨论了一下先不给

Comment (07akioni) [reply components/steps/index.en-US.md:42]: 之后再加吧，不是破坏性更新

**Comment by afc163** [comment]
样式偏移了 https://preview-25839-ant-design.surge.sh/components/steps-cn/#components-steps-demo-progress-dot

File: components/steps/index.tsx [hunk 4]
```diff
@@ -60,49 +50,36 @@ export default class Steps extends React.Component<StepsProps, any> {
...
-      if (status === 'process' && percentage !== undefined) {
+      if (status === 'process' && percent !== undefined) {
         // currently it's hard-coded, since we can't easily read the actually width of icon
         const progressWidth = size === 'small' ? 30 : 38;
         const iconWithProgress = (
           <div className={`${prefixCls}-progress-icon`}>
             <Progress
               type="circle"
-              percent={percentage}
+              percent={percent}
```
Comment (afc163) [inline components/steps/index.tsx:75]: 叫 currentPercent 会不会更准确一点？

Comment (07akioni) [reply components/steps/index.tsx:75]: 感觉必要不大？用户真的想改给他们 <Step> 上面改的机会吧，status 也没有叫 currentStatus

**Comment by 07akioni** [comment]
> 样式偏移了 https://preview-25839-ant-design.surge.sh/components/steps-cn/#components-steps-demo-progress-dot

我修一下

**Comment by zombieJ** [comment]
不紧贴了：
截屏2020-07-27 下午8 42 11

原：
截屏2020-07-27 下午8 42 21

**Review by zombieJ — APPROVED** [review]

**Pull request merged**